examples_tests/49.ComputeFFT/main.cpp

```diff
@@ -0,0 +1,629 @@
+	updateDescriptorSet_RemovePadding(driver, removePaddingDescriptorSet.get(), fftOutputBuffer_0, outImgView);
+	auto removePaddingDispatchInfo = getDispatchInfo_RemovePadding(outImageDim);
+
+	uint32_t outBufferIx = 0u;
+	auto lastPresentStamp = std::chrono::high_resolution_clock::now();
+	bool savedToFile = false;
+	
+	auto downloadStagingArea = driver->getDefaultDownStreamingBuffer();
+	
+	auto blitFBO = driver->addFrameBuffer();
+	blitFBO->attach(video::EFAP_COLOR_ATTACHMENT0, std::move(outImgView));
+
+	while (device->run() && receiver.keepOpen())
+	{
+		driver->beginScene(false, false);
+
+		// Src Image FFT X
+		driver->bindComputePipeline(fftPipeline_ImageInput.get());
+		driver->bindDescriptorSets(EPBP_COMPUTE, fftPipelineLayout_ImageInput.get(), 0u, 1u, &fftDescriptorSet_Src_FFT_X.get(), nullptr);
+		FFTClass::pushConstants(driver, fftPipelineLayout_ImageInput.get(), srcDim, paddedDim, FFTClass::Direction::X, false, FFTClass::PaddingType::CLAMP_TO_EDGE);
+		FFTClass::dispatchHelper(driver, fftDispatchInfo_Horizontal);
+
+		// Src Image FFT Y
+		driver->bindComputePipeline(fftPipeline_SSBOInput.get());
+		driver->bindDescriptorSets(EPBP_COMPUTE, fftPipelineLayout_SSBOInput.get(), 0u, 1u, &fftDescriptorSet_Src_FFT_Y.get(), nullptr);
+		FFTClass::pushConstants(driver, fftPipelineLayout_SSBOInput.get(), paddedDim, paddedDim, FFTClass::Direction::Y, false);
+		FFTClass::dispatchHelper(driver, fftDispatchInfo_Vertical);
+
+		// Convolution
+		driver->bindComputePipeline(convolvePipeline.get());
+		driver->bindDescriptorSets(EPBP_COMPUTE, convolvePipelineLayout.get(), 0u, 1u, &convolveDescriptorSet.get(), nullptr);
+		driver->pushConstants(convolvePipelineLayout.get(), nbl::video::IGPUSpecializedShader::ESS_COMPUTE, 0u, sizeof(uint32_t) * 3, &paddedDim); // pc.numChannels
+		driver->pushConstants(convolvePipelineLayout.get(), nbl::video::IGPUSpecializedShader::ESS_COMPUTE, sizeof(uint32_t) * 4, sizeof(uint32_t), &srcNumChannels); // numSrcChannels
+		dispatchHelper_Convolution(driver, convolveDispatchInfo);
+		
+		// Convolved IFFT X
+		driver->bindComputePipeline(fftPipeline_SSBOInput.get());
+		driver->bindDescriptorSets(EPBP_COMPUTE, fftPipelineLayout_SSBOInput.get(), 0u, 1u, &fftDescriptorSet_IFFT_X.get(), nullptr);
+		FFTClass::pushConstants(driver, fftPipelineLayout_SSBOInput.get(), paddedDim, paddedDim, FFTClass::Direction::X, true);
+		FFTClass::dispatchHelper(driver, fftDispatchInfo_Horizontal);
+		
+		// Convolved IFFT Y
+		driver->bindComputePipeline(fftPipeline_SSBOInput.get());
+		driver->bindDescriptorSets(EPBP_COMPUTE, fftPipelineLayout_SSBOInput.get(), 0u, 1u, &fftDescriptorSet_IFFT_Y.get(), nullptr);
+		FFTClass::pushConstants(driver, fftPipelineLayout_SSBOInput.get(), paddedDim, paddedDim, FFTClass::Direction::Y, true);
+		FFTClass::dispatchHelper(driver, fftDispatchInfo_Vertical);
```

Comment:
if you changed order of operation from 

```
FFT X
FFT Y
Convolve
iFFT X
iFFT Y
```
to
```
FFT X
FFT Y
Convolve
iFFT Y
iFFT X
```
then you could merge the middle 3 dispatches together